garden_ai/client.py

```diff
@@ -362,6 +364,114 @@ def register_pipeline(self, pipeline: Pipeline, container_uuid: str) -> str:
         local_data.put_local_pipeline(registered)
         return func_uuid
 
+    def _create_minimal_step(
+        self,
+        *,
+        model_name: str,
+        local_path: str,
+        flavor: str,
+        input_type: type,
+        output_type: type,
+        requirements: List[str],
```

Comment:
Having slept on it ... this is just too many parameters. And right below you bundle them up into this nice LocalModel object that is a logical grouping of fields that comprise an ML model. Could this method take a LocalModel? 

Please have another pass at this and think about how to use our existing Pydantic models or even come up with new ones as needed to make a more understandable interface that will take less maintenance and upkeep.